src/app/shared/book.service.ts

```diff
@@ -3,68 +3,100 @@ import { HttpClient, HttpErrorResponse } from "@angular/common/http";
-      return this.http.get(this.BaseUri+'/books/'+id).pipe(catchError(this.handleError)).toPromise();
-    }
+  getBooks() {
+    return this.http.get(this.BaseUri + '/books').pipe(catchError(this.handleError));
+  }
 
-    addBook(bookData: object) : Promise<Object> {
-      return this.http.post(this.BaseUri+'/books/addbook',bookData,{observe : 'response'}).pipe(catchError(this.handleError)).toPromise();
-    }
+  getCategories() {
+    return this.http.get(this.BaseUri + '/books/getcategories').pipe(catchError(this.handleError));
+  }
 
-    editBook(bookData : object) : Promise<Object> {
-      return this.http.put(this.BaseUri+'/books/'+bookData['bookId'],bookData,{observe : 'response'}).pipe(catchError(this.handleError)).toPromise();
-    }
+  findBookById(id: number): Promise<Object> {
+    return this.http.get(this.BaseUri + '/books/' + id).pipe(catchError(this.handleError)).toPromise();
+  }
+
+  addBook(bookData: object): Promise<Object> {
+    return this.http.post(this.BaseUri + '/books/addbook', bookData, { observe: 'response' }).pipe(catchError(this.handleError)).toPromise();
+  }
+
+  editBook(bookData: object): Promise<Object> {
+    return this.http.put(this.BaseUri + '/books/' + bookData['bookId'], bookData, { observe: 'response' }).pipe(catchError(this.handleError)).toPromise();
+  }
 
-    generateImageURL(title : string) : string {
-      var str = "assets/images/"+
-                title.split(' ').join('_').toLowerCase().concat(".jpeg");
-      return str;
+  generateImageURL(title: string): string {
+    var str = "assets/images/" +
+      title.split(' ').join('_').toLowerCase().concat(".jpeg");
+    return str;
+  }
+
+  generateDateString(jsonDate: object): string {
+    var str = "";
+    str = jsonDate['year'] + "-";
+    var month: string = jsonDate['month'] + "";
+    if (month.length == 1) {
+      month = "0" + month;
     }
-  
-    generateDateString(jsonDate : object) : string {
-      var str = "";
-      str=jsonDate['year']+"-";
-      var month : string = jsonDate['month']+"";
-      if(month.length==1) {
-        month="0"+month;
-      }
-      var day : string = jsonDate['day']+"";
-      if(day.length==1) {
-        day="0"+day;
-      }
-      str+=month+"-"+day;
-      return str;
+    var day: string = jsonDate['day'] + "";
+    if (day.length == 1) {
+      day = "0" + day;
     }
-  
-    private handleError(error: HttpErrorResponse) {
-      if (error.error instanceof ErrorEvent) {
-        // A client-side or network error occurred. Handle it accordingly.
-        console.error('An error occurred:', error.error.message);
-      } else {
-        // The backend returned an unsuccessful response code.
-        // The response body may contain clues as to what went wrong,
-        console.error(`Backend returned code ${error.status}, ` + `body was: ${error.error}`);
+    str += month + "-" + day;
+    return str;
+  }
+
+  addToCartService(book: Book) {
```

Comment:
Add to cart should be in Cart service instead of Book Service

Reply:
Yes done